File: C++/Problems/Problems/problem_2038.cpp
#include "problem_2038.h"

bool problem_2038::test() {
    std::string colors = "AAABABB";

    bool expected = true;

    auto result = winnerOfGame(colors);

    return expected == result;
}

bool problem_2038::winnerOfGame(std::string colors) {
    int n = colors.size();
    int aliceMoves = 0;
    int bobMoves = 0;
    for (int i = 1; i < n - 1; ++i)
    {
        if (colors[i - 1] == colors[i] && colors[i] == colors[i + 1])
        {
            if (colors[i] == 'A')
                aliceMoves++;
            else
                bobMoves++;
        }
    }

    return aliceMoves > bobMoves;
}

File: Python/Problems/Problems/problem2038.py
from itertools import repeat

from utils import print1DArray, print2DArray
from typing import List
from collections import Counter

from .problem import Problem


class Problem2038(Problem):
    def test(self) -> bool:
        colors = "AAABABB"

        expected = True

        result = self.winnerOfGame(colors)

        return expected == result

    def winnerOfGame(self, colors: str) -> bool:
        n = len(colors)
        aliceMoves = 0
        bobMoves = 0
        for i in range(1, n - 1):
            if colors[i - 1] == colors[i] == colors[i + 1]:
                if colors[i] == 'A':
                    aliceMoves += 1
                else:
                    bobMoves += 1

        return aliceMoves > bobMoves